sql/core/src/test/scala/org/apache/spark/sql/sources/BucketedReadSuite.scala

```diff
@@ -315,8 +317,14 @@ abstract class BucketedReadSuite extends QueryTest with SQLTestUtils {
         assert(
           joinOperator.right.find(_.isInstanceOf[SortExec]).isDefined == sortRight,
           s"expected sort in the right child to be $sortRight but found\n${joinOperator.right}")
+
+        if (expectedResult.isDefined) {
+          checkAnswer(joined, expectedResult.get)
```

Comment:
This will let us validate against some predefined expected output. 
L296 will recompute the result with the exact same `joinCondition`... but for my test case I don't want to reuse the same `joinCondition`